routers/web/repo/actions/view.go

```diff
@@ -236,6 +237,16 @@ func ViewPost(ctx *context_module.Context) {
 		Name: run.PrettyRef(),
 		Link: run.RefLink(),
 	}
+	refName := git.RefName(run.Ref)
+	if refName.IsBranch() {
+		b, err := git_model.GetBranch(ctx, ctx.Repo.Repository.ID, refName.ShortName())
+		if err != nil && !git_model.IsErrBranchNotExist(err) {
+			log.Error("GetBranch: %v", err)
+		} else if git_model.IsErrBranchNotExist(err) || (b != nil && b.IsDeleted) {
+			branch.Link = "" // remove the link of branch if it doesn't exist'
```

Comment:
is not better to give link to ref?

Reply:
2b6a605aec4c83c1ff1e9dfb056edff54fd17381